frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/transform.test.tsx

```diff
@@ -664,4 +666,275 @@ describe("Transforming mitxonline enrollment data to DashboardResource", () => {
       expect(filtered[0].b2b_organization_id).toBe(123)
     })
   })
+
+  describe("mitxonlineProgram", () => {
+    test("transforms V2Program to DashboardProgram correctly", () => {
+      const program = factories.programs.program()
+      const result = mitxonlineProgram(program)
+
+      expect(result).toEqual({
+        id: program.id,
+        key: `mitxonline-program-${program.id}`,
+        type: DashboardResourceType.Program,
+        title: program.title,
+        programType: program.program_type,
+        courseIds: program.courses,
+        collections: program.collections,
+        description: program.page.description,
+        reqTree: expect.any(Array),
+      })
+    })
+
+    test("transforms requirement tree correctly", () => {
+      const program = factories.programs.program({
+        req_tree: [
```

Comment:
https://github.com/mitodl/mit-learn/blob/77409934a9d06a3667c06b242c327aa07be8a773/frontends/api/src/mitxonline/test-utils/factories/requirements.ts#L20 might be useful for some of the tests